manyfews/calculations/zentra.py

```diff
@@ -0,0 +1,238 @@
+        #                       VP is the Vapour Pressure
+        #                       RH is the Relative Humidity between zero and one.
+
+        Tempair = zentraData["device"]["timeseries"][0]["configuration"]["values"][i][
+            3
+        ][7]["value"]
+
+        vapPressure = zentraData["device"]["timeseries"][0]["configuration"]["values"][
+            i
+        ][3][8]["value"]
+        rh = vapPressure / (0.611 * (math.exp((17.502 * Tempair) / (240.97 + Tempair))))
+
+        if rh > 1:
+            rh = 1
+        elif rh < 0:
+            rh = 0
+
+        RH.append(rh)
+
+    # output data sets into a dictionary
+    zentraDataSum = [
+        covertDate,
+        tStamp,
+        solar,
+        precip,
+        wDirection,
+        wSpeed,
+        gustSpeed,
+        vaporPressure,
+        atmosPressure,
+        airTem,
+        maxPrecipRate,
+        rhTemp,
+        VPD,
+        RH,
+    ]
+    zentraDataKeys = [
+        "local date",
+        "time stamp",
+        "solar radiation",
+        "precipitation",
+        "wind direction",
+        "wind speed",
+        "guest speed",
+        "vapor pressure",
+        "atmospheric pressure",
+        "air temperature",
+        "max precipitation Rate",
+        "RH senor temperature",
+        "VPD",
+        "Relative Humidity",
+    ]
+
+    zentraDataSet = dict(zip(zentraDataKeys, zentraDataSum))
+
+    return zentraDataSet
+
+
+def authZentraCloud():
+    """
+    This function is used to return authentication token for the Zentra cloud.
+    Username & password should be defined as environment parameters.
+
+    :return: authentication token.
+    """
+
+    # authentication into the Zentra Cloud API
+    token = ZentraToken(username=getenv("zentra_un"), password=getenv("zentra_pw"))
+
+    return token
+
+
+def dataBaseWriter(sn, backTime):
+    """
+    This function is used to save data from zentra cloud into
+    Database ( calculations_zentrareading table)
+    :param sn: met station Serial number
+    :param backTime: the back days number you want to extract (unit: day).
+    :return: none
+    """
+
+    # extract data from met station
+    zentraAtmos = zentraReader(backTime, sn, authZentraCloud())
+
+    # import data into DB
+
+    for i in range(len(zentraAtmos["local date"])):
+        zentraData = ZentraReading(
+            date=zentraAtmos["local date"][i],
+            device=zentraDevice,
+            precipitation=zentraAtmos["precipitation"][i],
+            relative_humidity=zentraAtmos["Relative Humidity"][i],
+            air_temperature=zentraAtmos["air temperature"][i],
+            wind_speed=zentraAtmos["wind speed"][i],
+            wind_direction=zentraAtmos["wind direction"][i],
+        )
+
+        zentraData.save()
+
+
+# get met station's SN
+zentraDevice = ZentraDevice.objects.get(device_sn="06-02047")
+sn = zentraDevice.device_sn
+
+# save into Data base
+backTime = 0.5
+dataBaseWriter(sn, backTime)
```

Comment:
Could this be a function with the serial number  and backTime as parameters? You could then add it as a task or document how to call it from `manage.py` shell.